csrc/multidevice/communicator.cpp

```diff
@@ -214,7 +214,9 @@ Communicator::Communicator(
 #endif
 }
 
-Communicator::~Communicator() {
+void Communicator::cleanup() {
+  store_ = nullptr;
```

Comment:
once we clear `store_` I don't think we currently have logic to restore it afterwards. Does that mean we need to check for `cleared()` status in `getBackendForTeam` to avoid accessing nullptr `store_`?

Reply:
Probably didn't make the comment clear.
I *think* we need to assert on `is_available` at the beginning of `getBackendForTeam` to detect pre-mature cleanup (or accidental cleanup from somewhere else).

But I'm happy to see that in the follow up PR along with the API change.

Reply:
Added a check in getBackendForTeam. I'm trying to avoid having to add an availability check in every API -- that wouldn't scale. Adding to getBackendForTeam makes sense because it's one method that's used by many APIs.